__tests__/invoices/invoice-details.spec.ts

```diff
@@ -0,0 +1,193 @@
+        invoiceType: InvoiceType.Lightning,
+        unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+      }
+
+      const invoiceDetails = createInvoiceDetails(usdAmountLightningParams)
+
+      expect(invoiceDetails).toEqual(
+        expect.objectContaining({
+          ...defaultExpectedInvoiceDetails,
+          invoiceType: InvoiceType.Lightning,
+          unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+          settlementAmount: { amount: 1, currency: WalletCurrency.Usd },
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            expiration: expect.any(Date),
+            invoiceDisplay: usdAmountInvoice,
+          }),
+        }),
+      )
+    })
+  })
+
+  describe("with btc receiving wallet", () => {
+    const btcWalletParams = {
+      ...defaultParams,
+      receivingWalletDescriptor: { id: "wallet-id", currency: WalletCurrency.Btc },
+    }
+
+    it("creates no amount lightning invoice", async () => {
+      const btcNoAmountLightningParams: CreateInvoiceDetailsParams<
+        typeof WalletCurrency.Btc
+      > = {
+        ...btcWalletParams,
+        invoiceType: InvoiceType.Lightning,
+      }
+
+      const invoiceDetails = createInvoiceDetails(btcNoAmountLightningParams)
+
+      expect(invoiceDetails).toEqual(
+        expect.objectContaining({
+          ...defaultExpectedInvoiceDetails,
+          invoiceType: InvoiceType.Lightning,
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            invoiceDisplay: noAmountInvoice,
+          }),
+        }),
+      )
+    })
+
+    it("create amount lightning invoice", async () => {
+      const btcAmountLightningParams: CreateInvoiceDetailsParams<
+        typeof WalletCurrency.Btc
+      > = {
+        ...btcWalletParams,
+        invoiceType: InvoiceType.Lightning,
+        unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+      }
+
+      const invoiceDetails = createInvoiceDetails(btcAmountLightningParams)
+
+      expect(invoiceDetails).toEqual(
+        expect.objectContaining({
+          ...defaultExpectedInvoiceDetails,
+          invoiceType: InvoiceType.Lightning,
+          unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+          settlementAmount: { amount: 1, currency: WalletCurrency.Btc },
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            invoiceDisplay: btcAmountInvoice,
+          }),
+        }),
+      )
+    })
+
+    it("creates an onchain address", async () => {
```

Comment:
```suggestion
    it("creates an btc onchain address", async () => {
```

we will probably have usd as well soon enough.

Reply:
`btc` is already in the name in the enclosing `describe` function.